packages/sitecore-jss/src/site/site-resolver.ts

```diff
@@ -0,0 +1,22 @@
+interface SiteData {
+  name: string;
+  hostName: string;
+  virtualFolder: string;
+  language: string;
+}
+
+// This resolver is used to determine site name based on the provided hostname
+export class SiteResolver {
+  /**
+   * Resolve siteName by hostName
+   * @param {string} hostName
+   * @param {SiteData[]} sitesInfo
+   * @param {string} [fallbackSiteName] siteName to be returned in case siteName is not found
+   * @returns {string} siteName
+   */
+  static resolve = (hostName: string, sitesInfo: SiteData[], fallbackSiteName?: string): string => {
```

Comment:
I was expecting the site resolver would perform the fetching of the `SiteData[]` as well (as opposed to a param) which should be possible once the `GraphQLSiteInfoService` is merged from [this PR](https://github.com/Sitecore/jss/pull/1227). However, I can see the benefit to keeping these separate.

What do you thing @art-alexeyenko? Probably a larger discussion we should have... Or something we can always adjust as POC moves along.

Reply:
I chose this approach, cause I thought that site-resolver can be used in multiple places

Reply:
I don't see much use in anything from SiteData/SiteInfo other than the site name, after it was resolved. Every other piece of data (hostname, language etc) that we may have there would already be known and serve as input into the resolve method.